### ISSUE TITLE
Using variable only in slice leads to false positive for F841: Locale variable is assigned but never used.

### ISSUE THREAD

**Issue by Wooza** [charliermarsh/ruff#1552]
Applying ruff on the following code leads to `repro.py:6:5: F841 Local variable 'shift_by' is assigned to but never used`:
```python
# File repro.py
from __future__ import annotations


def foo() -> None:
    shift_by = 1
    for a in list([1, 2])[shift_by:]:
        print(a)
```

Tested on ruff version `0.0.206`
